### PR TITLE
Datagen2017

### PR DESCRIPTION
**Pull request by RobertHSchmidt**

### CODE REVIEW

File: src/proto/diff.proto [hunk 1]
```diff
@@ -66,6 +66,15 @@ message ChangedFile {
 		CS4 = 204;
 		/** @exclude */
 		CS5 = 205;
+		
+		/** The file represents a JavaScript source file that had a parse error */
+		SOURCE_JS_ERROR = 300;
+		/** The file represents a JavaScript source file that parsed without error as JLS2 */
+		SOURCE_JS_JLS2 = 301;
```
Comment (psybers) [inline src/proto/diff.proto]: I dont think JavaScript is called 'JLS'.  It is defined by ECMAScript versions.  So the enums here should probably be named accordingly (SOURCE_JS_ECMA1/2/3/4/5/6/7 etc).  Also, which versions do we support?  Perhaps only include constants for those supported.

Comment (RobertHSchmidt) [inline src/proto/diff.proto]: The eclipse jsdt has two JLS fields under ast, JLS2 and JSL3. From the eclipse documentation JSL3 supports unto ECMA4 it is unclear what EMCA level JSL2 supports.
http://help.eclipse.org/kepler/index.jsptopic=%2Forg.eclipse.wst.jsdt.doc%2Freference%2Fapi%2Forg%2Feclipse%2Fwst%2Fjsdt%2Fcore%2Fdom%2FAST.html

Comment (psybers) [inline src/proto/diff.proto]: Yes, those field names are remnants of them using JDT as a starting point.  We should use proper names so as not to confuse people (JLS is the acronym known for Java Language Standard).  Their 'JLS3' is ECMAScript 4.  So assume the other one is version 3.

It appears the naming 'ES4' is somewhat standard.  I would go with that.

Also, I am a bit worried that the JSDT only supports up to ES4?  ES4 was abandoned and never completed.  It is also 10 years old and ES5, ES6, and ES7 are already out...

File: src/proto/diff.proto [hunk 2]
```diff
@@ -69,10 +69,10 @@ message ChangedFile {
 		
 		/** The file represents a JavaScript source file that had a parse error */
 		SOURCE_JS_ERROR = 300;
-		/** The file represents a JavaScript source file that parsed without error as JLS2 */
-		SOURCE_JS_JLS2 = 301;
 		/** The file represents a JavaScript source file that parsed without error as JLS3 */
-		SOURCE_JS_JLS3 = 302;
+		SOURCE_JS_ES3 = 301;
+		/** The file represents a JavaScript source file that parsed without error as JLS4 */
```
Comment (psybers) [inline src/proto/diff.proto]: Dont forget to update the comments.

**Comment by nguyenhoan** [comment]
Hi Robert,

It seems that Bob fixed all previous comment.
Do you have any other comments?

**Pull request merged**